### PR TITLE
Pal 1025 s3 k8s config

### PR DESCRIPTION
**Pull request by dev930018**

### CODE REVIEW

File: example-library/src/main/resources/application-example-perf.yaml [hunk 1]
```diff
@@ -11,52 +11,27 @@
...
-      "1-Pass through rule for no-policy directory": uk.gov.gchq.palisade.example.library.rule.PassThroughRule
-
-  serialiserProvider: std
-  serialisers:
-  - flavourFormat: ${default.format}
-    flavourType: ${default.type}
-    serialiserClass: uk.gov.gchq.palisade.example.library.serialise.AvroSerialiser
-
+# Locally, we use the SimpleResourceService which does a Files.walk(...) so no need to prepopulate
 # Used by the resource-service to infer the types of the (many thousands of) resources created for the performance test
```
Comment (pd104923) [inline example-library/src/main/resources/application-example-perf.yaml:35]:
```suggestion
# Used by the Resource Service to infer the types of the (many thousands of) resources created for the performance test
```

File: example-library/src/main/resources/application-example-s3.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+spring:
+  profiles:
+    include:
+    - example-k8s-config
```
Comment (pd104923) [inline example-library/src/main/resources/application-example-s3.yaml:18]:
```suggestion
    - example-s3-config
```

File: example-library/src/main/resources/application-example-s3.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    - example-users
+    - example-resources
+    - example-policies
+    - example-serialisers
+
+---
+
+spring:
+  profiles: example-k8s-config
```
Comment (pd104923) [inline example-library/src/main/resources/application-example-s3.yaml:27]:
```suggestion
  profiles: example-s3-config
```

**Review by pd104923 — CHANGES_REQUESTED** [review]
A few comments plus the licenses need to be reverted.

File: charts/deployment/templates/pre-install.yaml [hunk 4]
```diff
@@ -46,7 +46,9 @@ spec:
         command: [ "/bin/bash", "/bin/copyExampleData.sh" ]
         args:
           - {{ include "palisade.namespace" . | quote }}
+          - {{ .Values.global.deployment }}
           - {{ include "deployment.deployment.path" . | quote }}
+          - "/data/local-data-store"
```
[the remark this answers is not in the document]
Comment (dev930018) [reply charts/deployment/templates/pre-install.yaml:51]: Not in helm, that's all java problems

[the remark this answers is not in the document]
Comment (dev930018) [reply charts/deployment/templates/pre-install.yaml:51]: I'll add one for clarity though

**Review by pd104923 — APPROVED** [review]

**Review by developer01189998819991197253 — CHANGES_REQUESTED** [review]
There are also a lot of dependency tree changes in this PR but no pom version changes, are these from previous updates that have not been built and therefore updated dependency trees?

File: deployment/src/main/resources/k8s-bash-scripts/copyExampleData.sh [hunk 5]
```diff
@@ -13,39 +13,52 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Example usage
+# ./copyExampleData.sh default example /mnt/shared-classpath-jars /data/local-data-store
+
+set -e
```
[the remark this answers is not in the document]
Comment (developer01189998819991197253) [reply deployment/src/main/resources/k8s-bash-scripts/copyExampleData.sh:19]: bump

File: deployment/src/main/resources/k8s-bash-scripts/copyExampleData.sh [hunk 6]
```diff
@@ -13,39 +13,52 @@
...
+
 # Copy example AVRO data that will be read by the data-service
 # The location of these is dependant on the resource-service pre-population values
-echo "Preserving previous run data for $DATASTORE"
-ls -R $DATASTORE
-cp -vrf /usr/share/example-data/resources/data/* $DATASTORE
-echo "Copied all example-data to $DATASTORE"
-ls -R $DATASTORE
+echo "Preserving previous run data-files for $DATASTORE"
+ls -R "$DATASTORE"
```
[the remark this answers is not in the document]
Comment (developer01189998819991197253) [reply deployment/src/main/resources/k8s-bash-scripts/copyExampleData.sh:54]: bump

File: licenses/mit_license.html [hunk 7]
```diff
@@ -18,7 +18,7 @@
 <link type="text/css" rel="stylesheet" href="https://opensource.org/files/css/css_7rptTjxTlnnwcQOT_3B6dmWJfzo0pnNKsCvuv3HArrk.css" media="all" />
 
 <!--[if (lt IE 9)]>
-<link type="text/css" rel="stylesheet" href="https://opensource.org/sites/all/themes/bootstrap-business/css/ie8.css?qop0k3" media="all" />
```
[the remark this answers is not in the document]
Comment (developer01189998819991197253) [reply licenses/mit_license.html]: bump

**Review by developer01189998819991197253 — CHANGES_REQUESTED** [review]

**Review by pd104923 — DISMISSED** [review]

**Review by pd104923 — APPROVED** [review]

**Review by developer01189998819991197253 — APPROVED** [review]

**Pull request merged**